### PR TITLE
Config fixes for Circle deployments: skip-maint and refresh-db

### PR DESCRIPTION
**Pull request by weitzman**
**Jira:**
https://jira.mass.gov/browse/DP-16585

**To Test:**
- [x] [Verify that we have a successful hotfix workflow](https://circleci.com/workflow-run/6c66abcc-2541-4edd-9a66-a0e4acf566d6)

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -664,6 +676,7 @@ workflows:
             tags:
               only: /.*/
       - deploy:
+          name: deploy-test-no-refresh-db
```
Comment (yriahi) [inline .circleci/config.yml]:
```suggestion
          name: deploy-tag-stage-acquia
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -686,10 +699,11 @@ workflows:
             tags:
               only: /.*/
       - deploy:
+          name: deploy-prod-no-refresh-db
```
Comment (yriahi) [inline .circleci/config.yml]:
```suggestion
          name: deploy-tag-prod-acquia
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -686,10 +699,11 @@ workflows:
...
           requires:
             - build
             - tagging_acquia
-            - deploy
+            - deploy-test-no-refresh-db
```
Comment (yriahi) [inline .circleci/config.yml]:
```suggestion
            - deploy-tag-stage-acquia
```

**Review by yriahi — CHANGES_REQUESTED** [review]
@weitzman Looks good to me with some minor renamings. 
Please see the suggested changes.

File: .circleci/config.yml (hunk 1 above)
Comment (weitzman) [reply .circleci/config.yml]: Jobs always run in the context of a workflow in Circle 2.1 so think its unnecesarily verbose to put the workflow name in each job name. i mean, its not in the build job's name, for example.

Comment (yriahi) [reply .circleci/config.yml]: Fine by me.

**Pull request merged**